src/tape_drivers/linux/sg/sg_tape.c

```diff
@@ -392,6 +395,14 @@ static int _take_dump(struct sg_data *priv, bool capture_unforced)
 	time_t    now;
 	struct tm *tm_now;
 
+	/* To check if the function became recursive */
+	static unsigned char recursive_counter = 0;
```

Comment:
`static` variable is shared with all threads. So this is problematic when it is used in the multiple drive environment.

You need to use local variable or define this variable into `struct sg_data`.